### PR TITLE
Add paper on internals

### PR DESCRIPTION
**Pull request by neithernut**
This patchset will add the basic lineout of a paper which describes the datastructure the library uses internally.

### CODE REVIEW

File: papaer/main.tex [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+\input{intro}
+
+\input{overview}
+
+\input{data-layout}
+
+\input{operations}
+
```
Comment (matthiasbeyer) [inline papaer/main.tex]: Remove the blank lines between the `\input` commands, please.

File: papaer/main.tex [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+\input{parameters}
+
+
+\end{document}
+
+
+
+
```
Comment (matthiasbeyer) [inline papaer/main.tex]: There are much too much blank lines at the end of the file.

File: papaer/intro.tex [hunk 3]
```diff
@@ -0,0 +1,27 @@
+\section{Introduction}
+
+    %TODO: definition of "set"
+
+    %TODO: refs to some standard source on data structures
+    ``Sets'' or ``Maps'' are often implemented using either hash tables or AVL
+    trees.  For the former one, insertion, deletion and query of an element via
+    a key are generally in $O(1)$ in the average case, but may be in $O(n)$ in
+    the worst case.  For AVLs, these operations are in $O(ln(n))$ in the average
+    \em and\em{} in the worst case. If the usage of a set is restricted to these
```
Comment (matthiasbeyer) [inline papaer/intro.tex]: I think you can use `\em{and}` for single words, can't you? Would be cleaner.

File: papaer/overview.tex [hunk 4]
```diff
@@ -0,0 +1,17 @@
+\section{Overview}
+
+    The proposed data-structure is a hash-table which uses an amortised dynamic
+    array for it's list of buckets and AVLs as implementation for it's buckets.
```
Comment (matthiasbeyer) [inline papaer/overview.tex]: Typo: Write "for its list"  (remove the `'`)

Comment (matthiasbeyer) [inline papaer/overview.tex]: Same at the end of the line.

File: papaer/overview.tex [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    While each bucket is a fully functional set on it's own, splitting them into
```
Comment (matthiasbeyer) [inline papaer/overview.tex]: And here again, remove the `'`.

File: papaer/overview.tex [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    smaller domains with a maximum average number of elements reduces the
+    average complexity for queries, insertions and other operations.
+
+    This paper will discuss the algorithms used to implement each operation,
+    drawing lines between the different levels, the overall hash-table and the
+    buckets that is. Each operation's description will also include a short
+    derivation of the operation's class of complexity.
```
Comment (matthiasbeyer) [inline papaer/overview.tex:12]: I think you should remove the `'` here to and in the line above.

**Comment by matthiasbeyer** [comment]
Thank you for your PR. Please fix the proposed stuff, and rewrite the commit message for the first commit (ee5f438),  which has a typo.

Please rebase and squash/fixup the commits, to keep the history clean.

File: papaer/overview.tex (hunk 6 above)
Comment (neithernut) [inline papaer/overview.tex:12]: Hi,

> I think you should remove the `'` here to and in the line above.
> Nope, it's "of the operation", hence "the operation's" (possesive).

Regards,
Jule

File: papaer/intro.tex (hunk 3 above)
Comment (neithernut) [inline papaer/intro.tex]: Hi,

> I think you can use `\em{and}` for single words, can't you? Would be
> cleaner.
> It's "\emph{and}" in this case.

Regards,
Jule

**Comment by neithernut** [comment]
Hi,

> Please rebase and squash/fixup the commits, to keep the history clean.
> All things which ought to be fixed are fixed now, I think.

Regards,
Jule

**Pull request merged**